wiki/python/ESTIMATOR_GUIDE.md

```diff
@@ -0,0 +1,709 @@
+# cuML Python Estimators Developer Guide
+
+This guide is meant to help developers follow the correct patterns when creating/modifying any class that inherits from `cuml.common.base.Base`.
+
```

Comment:
It would be nice if we can start with a high-level guide: ```

A cuML Estimator must:
 * Inherit from cuml.common.Base
 * Follow the scikit-learn estimator guidelines (link)
 * Declare each array-type attribute they will compute as a class variable so they can be automatically converted to appropriate types
 * Add input type and return type annotations to public API functions OR wrap those functions explicitly with conversion decorators (see ...)
```

I think as the guide gets longer, it can be easy to get lost in it.

Reply:
I actually ended up reorganizing it more than intended. I liked your suggestion of moving the more advanced details to the end for both the decorators and descriptors. Allows dev's wanting a quick intro less of a read without removing any information.

Would be great if you could glance over the new organization and let me know what you think.